### ISSUE TITLE
@types/react should be listed in peerDependencies

### ISSUE THREAD

**Issue by theKashey** [zzarcon/ts-react-toolbox#20]
https://github.com/gaearon/react-hot-loader/issues/1359

> However, you don't specify the dependency for @types/react, which causes issues in projects where TypeScript compiler is running in "strict" mode and where dependencies are installed by a more strict package manager like pnpm (Rush).

> It gives the following TypeScript error during application compilation:

```ERROR in ~/app/common/temp/node_modules/.registry.npmjs.org/react-hot-loader/4.12.15_react-dom@16.10.2+react@16.10.2/node_modules/react-hot-loader/index.d.ts(1,24):

TS7016: Could not find a declaration file for module 'react'.
'~/app/common/temp/node_modules/.registry.npmjs.org/react/16.10.2/node_modules/react/index.js' implicitly has an 'any' type.
```

**Comment by zzarcon** [comment]
Good point. Is this only affecting peer deps? Just wondering what happens if you use or re expose a type from your lib, should that be listed as well?

**Comment by theKashey** [comment]
Probably no. You shall only "depend" on it, so the package manager would understand it.
